lib/abide_dev_utils/xccdf.rb

```diff
@@ -218,6 +229,19 @@ def controls_by_profile_title(profile_title)
         profiles.select { |x| x.title == profile_title }.controls
       end
 
+      def gen_map(opts)
+        os, ver = facter_platform
+        if opts[:dir]
+          mapping_dir = File.expand_path(File.join(opts[:dir], opts[:type], os, ver))
+        else
+          mapping_dir = ''
+        end
+        ['title', 'hiera_title', 'hiera_title_num', 'number'].each_with_object({}) do |idx, h|
+          map_file_path = "#{mapping_dir}/#{idx}.yaml"
+          h[map_file_path] = map_indexed(index: idx, framework: opts[:type], key_prefix: opts[:parent_key_prefix])
```

Comment:
Are there issues caused by `opts[:parent_key_prefix]` not existing an subsequently passing `nil` to the `map_indexed` function?

Reply:
That possibility is handled by cli/xccdf.rb:46
@data[:parent_key_prefix] = '' if @data[:parent_key_prefix].nil?

Reply:
What happens if I try and use the `gen_map` method from a different context than the CLI, though?

Reply:
The gen_map method doesn't write anything.  That's handled in CLI.

Reply:
I understand, but it does return a hash. What I'm asking is what happens if I want to use that method in a script or something and that value is `nil`?

Reply:
You get a stack trace.

Reply:
Shouldn't that be a required for `gen_map` then because the method doesn't work without a value there?

Reply:
Well it's functionally required...  Does ruby have some way to specify it's required?

Reply:
yeah, you just make a bare parameter declaration in the method signature. So instead of `gen_map(opts)` it would be `gen_map(parent_key_prefix, opts)` and then you would pass the parent key prefix in explicitly. Another option would be to write the method signature with keyword params like this: `gen_map(dir: <default val>, type: <default val>, parent_key_prefix: <default val>)` and then you could call it with the opts hash using a double-splat: `gen_map(**@data)` which will break the hash out into it's individual key-value pairs and match those up with the keyword args.

Reply:
Ok, so I change line 233 to `def gen_map(parent_key_prefix, opts)`. But if you pass in nil for parent_key_prefix you still get the same stack trace.   ...I'm not sure what this accomplishes.

Reply:
That's because you're explicitly passing `nil`. Bare params just mean you must pass something to them.